drivers/include/periph/freqm.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2023 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_periph_freqm FREQM
+ * @ingroup     drivers_periph
+ * @brief       FREQM peripheral driver interface
+ *
+ * This interface allows to configure and use the for Frequency Meter (FREQM)
+ * available on some Microchip Boards.
+ *
+ * The Frequency Meter uses the frequency of a known reference clock to
+ * determine with a 24-bit accuracy the frequency of another clock.
```

Comment:
Is this purely internal? I thought this was to measure the frequency of a signal on a GPIO.

Reply:
> Is this purely internal? I thought this was to measure the frequency of a signal on a GPIO.

It's not purely internal, the clocks can be driven by I/O pins. To claify this we could write something like
```suggestion
 * determine with a 24-bit accuracy the frequency of another clock.
 * These clocks can be driven by I/O pins.
```

Reply:
I actually can't think of a use case for measuring internal clocks. I implemented your suggestion to supply a GPIO instead.